tests/dbz-twin/rac/db-check.py

```
@@ -0,0 +1,377 @@
+    return False
+
+
+def replay_events(conn, source_table):
+    """Replay CDC events into final row state.
+
+    Skips SEED and SENTINEL events. Returns: {(table_name, pk): {col_upper: value, ...}}
+    """
+    state = {}
+    for row in conn.execute(
+        f"SELECT table_name, op, raw_json, event_id FROM {source_table} "
+        "ORDER BY event_id, seq"
```

Ruff (0.15.9)

[error] Possible SQL injection vector through string-based query construction

(S608)